### DISCUSSION TITLE
is it possible to hide warnings

### DISCUSSION THREAD

**Discussion by teto** [starship/starship#1833] [Q&A] [answered]
I use nix and as such I have almost nothing in my PATH. Every project gets its own `virtual` environment (called nix-shell).
The issue is that starship detects a python project and thus triggers:
```
[WARN] - (starship::utils): Executing command "python" failed by: Os { code: 2, kind: NotFound, message: "No such file or directory" } 
```
Indeed python is not available until I start my nix-shell. Same for node etc. Is there a way to hide those messages ?

**Comment by davidkna** [comment, marked as answer]
The log level for that message should have already been reduced to no longer be visible by default in starship 0.46.0 or higher  (#1730).

If you still wish to hide log messages you could try setting the `STARSHIP_SESSION_KEY` environment variable to a persistent value in your shell profile after starship has been initialised.

**Reply by teto** [reply]
than you I have upgraded and it is fixed
